tokio/src/macros/cfg.rs

```diff
@@ -154,13 +155,20 @@ macro_rules! cfg_not_io_util {
 
 macro_rules! cfg_loom {
     ($($item:item)*) => {
-        $( #[cfg(loom)] $item )*
+        $(
+            #[cfg(loom)]
+            #[cfg(not(target_os = "wasi"))]
+            $item
+        )*
```

Comment:
We can just not compile loom tests on WASM rather than disabling them like this.